### PR TITLE
Tidier column names and types

### PR DESCRIPTION
**Pull request by mstrimas**
Convert column names to `lower_snake_case` to avoid users having to employ backticks. Eventually the EBD will be released with a header using the same column name format. Only way I was able to figure out how to do this was via ` dplyr::select(ds, dplyr::all_of(col_names))`, but very open to a simpler way if there is one.

Set column types based on name rather than position. Only those columns that aren't character are specified, everything else defaults to character. This has three benefits:
1. The EBD has been know to change without warning, e.g. dropping or adding columns or changing order, so hopefully this protects us against that. 
2. Many of the EBD columns are not required or redundant, and users may want to remove columns prior to importing to parquet to reduce file size.
3. This will allow us to now import either the EBD of Sampling Event Data using the same code since the SED has a subset of the EBD columns.

### CODE REVIEW

**Comment by cboettig** [comment]
Very nice.  Yup, totally with you on `snake_case`. I was hesitant to change from the default heading but happy to do so with your endorsement!  Also agree on the explicit column names, thanks for adding that.

**Comment by cboettig** [comment]
Oh minor thing @mstrimas , can you update the unit test to use the new snake_case column name?  See checks above.  (yeah so far the unit testing is pretty minimal!)

**Comment by mstrimas** [comment]
Oops, forgot about those unit tests, should be fixed now!

File: R/import_ebird.R [hunk 1]
```diff
@@ -13,47 +13,75 @@
...
+  ds <- dplyr::select(ds, dplyr::all_of(col_names))
+  
   # Consider alternative partitions that might speed common queries
   # partitioning = c("COUNTRY")
   # However, arrow supports at most 1024 partitions
   
-  arrow::write_dataset(ds, dest, format="parquet")
+  arrow::write_dataset(ds, dest, format = "parquet")
   
+  unlink(source_dir, recursive = TRUE)
```
Comment (mstrimas) [inline R/import_ebird.R:39]: Meant to put this in a separate PR, but just realized deleting the temp txt.gz file as per Issue #3 ended up in this PR. Do you want me to separate it out or OK to just leave it here.

**Comment by cboettig** [comment]
Nice, thanks!

I'm fine with having the patch for #3 in here.   (As long as the PR is small enough to enable easy review I never mind addressing multiple issues in a single PR.  I think that kind of modularity is more important only on bigger / widely used projects)
